src/wallet/coinselection.cpp

```diff
@@ -105,8 +115,14 @@ std::optional<SelectionResult> SelectCoinsBnB(std::vector<OutputGroup>& utxo_poo
             // value. Adding any more UTXOs will be just burning the UTXO; it will go entirely to fees. Thus we aren't going to
             // explore any more UTXOs to avoid burning money like that.
             if (curr_waste <= best_waste) {
-                best_selection = curr_selection;
-                best_waste = curr_waste;
+                // Don't add the selection if the max weight was exceeded
+                if (curr_selection_weight > max_weight) {
+                    max_tx_size_exceeded = true; // at least one result exceeds the max weight
+                } else {
+                    // Set best selection
+                    best_selection = curr_selection;
+                    best_waste = curr_waste;
+                }
```

Comment:
We should not continue searching a subtree when the selection already exceeds the `max_weight`. Instead of only checking it when we have selected sufficient funds, I would add `curr_selection_weight > max_weight` as a backtrack reason before `else if (curr_value >= selection_target)`:

        if (curr_value + curr_available_value < selection_target || // Cannot possibly reach target with the amount remaining in the curr_available_value.
            curr_value > selection_target + cost_of_change || // Selected value is out of range, go back and try other branch
            (curr_waste > best_waste && is_feerate_high)) { // Don't select things which we know will be more wasteful if the waste is increasing
            backtrack = true;
        } else if (curr_selection_weight > max_weight) { // Exceeding weight for standard tx, cannot find more solutions by adding more inputs
            max_tx_size_exceeded = true; // at least one selection attempt exceeded the max weight
            backtrack = true;
        } else if (curr_value >= selection_target) {       // Selected value is within range

Reply:
yeah better 👌🏼 , thanks